src/v/kafka/server/connection_context.h

```diff
@@ -331,6 +336,8 @@ class connection_context final
     std::optional<security::tls::mtls_state> _mtls_state;
     config::binding<uint32_t> _max_request_size;
     ss::lowres_clock::time_point _throttled_until;
+    config::binding<std::unordered_set<int16_t>>
+      _kafka_throughput_controlled_api_keys;
```

Comment:
I think that the representation at runtime should be something more efficient. at the very least a flat hash set. but some other options to consider:

* std::vector<bool>
* std::bitset
* roaring bitset (we have this in the tree already)

especially i think the vector of bool may be small enough (with the specialization implementation) to fit in a cache line.